content/docs/reference/design-docs/editoast-errors/_index.en.md

```diff
@@ -0,0 +1,320 @@
+---
+title: "Editoast error management"
+linkTitle: "Editoast error management"
+weight: 80
+---
+
+# Goals
+
+- Have a clear separation between logically distinct errors.
+- Dispose of a way to actually match on errors when they occur deeper in the stack
+- Tie the errors to the endpoint they originate from in the OpenAPI.
+- Separate the error definition and emission of their serailization.
+- Statuate on how we want to forward Core's errors.
+- End the `InternalError` hegemony ⚔️
+
+# Constraints
+
+- Keep the same error format.
+- The error must live until the error is handled, conversion to our standard error format only happen in the response serialization.
+- Errors must implement `std::error::Error`.
+- Errors must be composable.
+- Errors variants must be shareable (e.g.: no multiple `InfraNotFound` variants).
+- Each endpoint must provide all its error cases in the OpenAPI. (How the frontend will consume them is another problem that we'll have to deal with.)
+- The `context` field of the error is populated in the views.
+- Errors can be handled in a generic matter (for situations where it makes some sense to do so).
+  - I.e.: some form of downcasting is available.
+
+# New system
+
+- Rely on `thiserror` everywhere.
+- Keep the `trait EditoastError` but only implement it for errors defined in `views`.
+- Rework `derive(EditoastError)` to interface it with `derive(thiserror::Error)`.
```

Comment:
We could live without it, but having it would make assigning eg. error codes much simpler.

Reply:
We need for each error:

* `impl EditoastError for MyError` (moderately tedious to write by hand)
* `impl utoipa::ToSchema for MyError` (absolutely **insane** to write by hand)

If that's not convincing enough, manually implementing `EditoastError for` would really hinder the readability of the type. With the macro, all information about an error is gathered around its variant, but it will be scattered across all associated functions in the `impl` block.

I believe we're well above the threshold of acceptable boilerplate to justify the complexity of a proc macro ^^